File: redsheet-portal/src/app/database-models/their-batna.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectRequirement,  } from './'
export class TheirBatna {
    constructor(theirBatna?: TheirBatna )  {
        let properties = [ "theirBatanValue", "theirBatnaId", "projectRequirementId",];
        for (let property of properties)
            if (theirBatna && theirBatna[property])
                this[property] = theirBatna[property];
    }
 
    @required()
    @maxLength(400)
	theirBatanValue : string =   undefined;
 
	theirBatnaId : number =   0 ;
 
    @range(0,2147483647)
	projectRequirementId : number =   undefined;
	projectRequirement : ProjectRequirement  ;


}

File: RedSheet.DbEntities/Models/TheirBatna.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("TheirBatnas", Schema = "dbo")]
	[Serializable]
	public partial class TheirBatna  
	{
 
	[Required] 
	[MaxLength(400)]
	public string TheirBatanValue { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int TheirBatnaId { get; set; } 
 
	[RelationshipTableAttribue("ProjectRequirements","dbo")]
	//Relationships
	public int ProjectRequirementId { get; set; } 
 
	[ForeignKey("ProjectRequirementId")]
	public ProjectRequirement ProjectRequirement { get; set; } 

	public TheirBatna()
	{
    }

    }
}

